### PR TITLE
Update release notes for Network Operator v24.7.0

### PR DESCRIPTION
**Pull request by e0ne**

### CODE REVIEW

**Comment by rollandf** [comment]
@kvadrage PTAL

File: docs/platform-support.rst [hunk 1]
```diff
@@ -164,13 +164,13 @@ NVIDIA Network Operator has been validated in the following scenarios:
      - 
    * - Red Hat Core OS
      - 
-     - 4.12-4.15
+     - 4.12-4.16
```
Comment (rollandf) [inline docs/platform-support.rst:167]: 4.14-4.16

This is what we use in the OCP bundle:
https://github.com/Mellanox/network-operator/blob/master/Makefile#L50

Comment (noama-nv) [reply docs/platform-support.rst:167]: supporting 3 releases

Comment (rollandf) [reply docs/platform-support.rst:167]: @noama-nv so, what should we have here?

File: docs/release-notes.rst [hunk 2]
```diff
@@ -36,6 +36,11 @@ Changes and New Features
 
    * - Version
      - Description
+   * - 24.7.0
+     - | - Added support for OpenShift Container Platform v4.16.
```
Comment (noama-nv) [inline docs/release-notes.rst:40]: @e0ne 
add:
Support for RHEL with containerd and DOCA-OFED container deployment
Support Switchdev SRIOV mode with OVS CNI - Externally managed PF/Bridge - Testing only
Switchdev VF config restoration for OVS-CNI in MOFED/DOCA container

File: docs/platform-support.rst (hunk 1 above)
Comment (e0ne) [reply docs/platform-support.rst:167]: aligned with Makefile

File: docs/release-notes.rst (hunk 2 above)
Comment (e0ne) [reply docs/release-notes.rst:40]: done

**Review by rollandf — APPROVED** [review]

**Pull request merged**